### PR TITLE
support setting ImageFamily and BootMode when building image

### PR DESCRIPTION
**Pull request by hwhuangzsly**
support setting ImageFamily and BootMode when building image

### CODE REVIEW

**Review by lbajolet-hashicorp — COMMENTED** [review]
Hi @hwhuangzsly,

Thanks for the PR, I left a suggestion regarding validating the image family's contents before doing the build, may I also suggest adding unit tests for the config? It would be useful to make sure the invalid cases are caught during Prepare.

I'll let you address those, feel free to ping me when you're done (or if you have questions), and I'll do another pass then.

Thanks again!

**Comment by hwhuangzsly** [comment]
@lbajolet-hashicorp Thanks for your suggestions. I have added tests to the "target_image_family" and "boot_mode", and optimize the verification method and description of the "target_image_family".  Please take a review again~

File: builder/ecs/image_config_test.go
Comment (lbajolet-hashicorp) [inline builder/ecs/image_config_test.go:110]: Just to be sure the `\p{L}` expression captures that, mind adding a test with chinese characters since they're explicitly documented as valid for the URIs?

File: builder/ecs/image_config.go
Comment (lbajolet-hashicorp) [inline builder/ecs/image_config.go:195]: I don't know all my unicode character classes by heart, but doesn't L also match the `a-zA-Z` ranges? We can probably simplify that expression if that's the case
```suggestion
			imageFamilyReg := regexp.MustCompile(`^\p{L}[\p{L}_0-9\-\.\:]{1,127}$`)
```
I believe also that both `.` and `:` don't need escaping in the context of a `[]`? Not sure though, that probably depends on the regex engine.

**Review by lbajolet-hashicorp — APPROVED** [review]
Hey @hwhuangzsly,

Thanks for the reroll! I've left only a couple of comments left on the regex and the tests, but overall this looks good to me, I'll let you address those and we can come back for another round of reviews then.

Pre-approving so it's not a problem later, the code is near-ready I think.

File: builder/ecs/image_config_test.go
Comment (hwhuangzsly) [reply builder/ecs/image_config_test.go]: I add new tests to ensure the regex can match Chinese characters, -, _, : and .

File: builder/ecs/image_config.go
Comment (hwhuangzsly) [reply builder/ecs/image_config.go]: After testing, I find that `p{L}` can match `a-zA-Z` but can't match `.` or digit, so I simplify it as your suggestion. It's pleasure for me to learn more about the regex rule.

**Comment by lbajolet-hashicorp** [comment]
The last reroll looks good to me @hwhuangzsly, thanks for pushing this one forward!

I'll wait for tests to go green and merge this.

**Comment by hwhuangzsly** [comment]
@lbajolet-hashicorp Excuse me, do you know when the next release version will be made?

**Comment by lbajolet-hashicorp** [comment]
Hi @hwhuangzsly,

This can happen reasonably soon I would think, maybe today or tomorrow.

I'll let you know when this happens

**Comment by hwhuangzsly** [comment]
@lbajolet-hashicorp Excuse me, as we previously mentioned,  the next release version will be made soon. I think maybe you forget it, just a gentle reminder.

**Comment by lbajolet-hashicorp** [comment]
Hi @hwhuangzsly,

Thanks for the reminder, I did forget indeed.
I've opened a PR for starting this release, provided this is accepted and merged quickly, the release will follow soon after.

**Pull request merged**
